src/main/treeDataProviders/utils/analyzerUtils.ts

```diff
@@ -186,57 +198,150 @@ export class AnalyzerUtils {
+     * @param applicableTimeStamp - ended time stamp for the applicable scan
+     * @param elapsedTimeInMillsSec - elapsed time that took the scan to run
+     */
+    private static transferApplicableResponseToBundles(
+        applicableIssues: ApplicabilityScanResponse,
+        bundles: Map<FileScanBundle, Set<string>>,
+        logManager: LogManager,
+        applicableTimeStamp: number,
+        elapsedTimeInMillsSec: number
+    ) {
+        for (let [bundle, relevantCve] of bundles) {
+            let descriptorIssues: DependencyScanResults = <DependencyScanResults>bundle.data;
+            // Filter only relevant information
+            descriptorIssues.applicableScanTimestamp = applicableTimeStamp;
+            descriptorIssues.applicableIssues = AnalyzerUtils.filterOnlyRelevantApplicableData(applicableIssues, relevantCve);
+            // Populate it in bundle
             let applicableIssuesCount: number = AnalyzerUtils.populateApplicableIssues(
-                fileScanBundle.root,
-                fileScanBundle.dataNode,
+                bundle.root,
+                <ProjectDependencyTreeNode>bundle.dataNode,
                 descriptorIssues
             );
-            scanManager.logManager.logMessage(
+            logManager.logMessage(
                 'Found ' +
                     applicableIssuesCount +
                     " applicable CVE issues in descriptor = '" +
                     descriptorIssues.fullPath +
                     "' (elapsed " +
-                    (Date.now() - startApplicableTime) / 1000 +
+                    elapsedTimeInMillsSec / 1000 +
                     ' seconds)',
                 'INFO'
             );
-            fileScanBundle.root.apply();
+            bundle.root.apply();
         }
     }
 
+    /**
+     * Filter a given full applicable data to only relevant information base on a given cve list
+     * @param applicableIssues - all the applicable information
+     * @param relevantCve - cve list to filter information only for them
+     * @returns applicableIssues with information relevant only for the given relevantCve
+     */
+    private static filterOnlyRelevantApplicableData(
+        applicableIssues: ApplicabilityScanResponse,
+        relevantCve: Set<string>
+    ): ApplicabilityScanResponse {
+        let scanned: string[] = [];
+        let allApplicable: Map<string, CveApplicableDetails> = new Map<string, CveApplicableDetails>(Object.entries(applicableIssues.applicableCve));
+        let applicable: Map<string, CveApplicableDetails> = new Map<string, CveApplicableDetails>();
+
+        for (let scannedCve of applicableIssues.scannedCve) {
+            if (relevantCve.has(scannedCve)) {
+                scanned.push(scannedCve);
+                let potential: CveApplicableDetails | undefined = allApplicable.get(scannedCve);
+                if (potential) {
+                    applicable.set(scannedCve, potential);
+                }
+            }
+        }
```

Comment:
The current implementation of the `filterOnlyRelevantApplicableData` function is not very clear in terms of its purpose and contains ambiguous variable names, making it difficult to understand from the reader's perspective